Frontend/src/components/SignUpPage.jsx

```diff
@@ -1,43 +1,122 @@
-import {createUserWithEmailAndPassword} from "firebase/auth";
-import { useState} from 'react';
-import {auth} from "../firebase";
-import "./SignUpPage.css"
-import AuthDetails from "./AuthDetails";
-
-function SignUpPage() {
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
-
-    const signUp = (e) => {
-        e.preventDefault();
-        createUserWithEmailAndPassword(auth, email, password)
-        .then((userCredentials) => {
-            console.log(userCredentials)
-        }).catch((error) => {
-            console.log(error)
-        })
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignUpPage.css'
+import { UserContext } from '../UserContext.jsx';
+
+const SignUpPage = () => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [userRole, setUserRole] = useState('');
+
+  const { updateUser } = useContext(UserContext);
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      // Make the signup API request
+      const response = await fetch(`http://localhost:3000/users/signup`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ name, email, password, userRole }),
+        credentials: 'include'
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        const loggedInUser = data.user;
+
+        console.log('Signup successful');
```

Comment:
For the production code we don't want console.log in it, you can either remove it or add a todo to clean up later

Reply:
Resolved